clusterman/batch/node_migration.py

```diff
@@ -113,10 +136,12 @@ def _spawn_worker(self, label: WorkerProcessLabel, routine: Callable, *args, **k
             self.logger.warning(f"Worker labelled {label} already running, skipping")
             return False
         running_workers = sum(proc.is_alive() for proc in self.migration_workers.values())
-        if isinstance(label, MigrationEvent) and running_workers >= self.available_worker_slots:
+        if self._is_event_worker_label(label) and running_workers >= self.available_worker_slots:
             # uptime workers are prioritized skipping this check
             self.logger.warning(f"Too many worker processes running already ({running_workers}), skipping")
             return False
+        lock_key = label.split(self.WORKER_LABEL_SEPARATOR, 1)[1]
```

Comment:
i'm personally a fan of avoiding explicit indexing, but that might just be a me thing and i guess
```python
        _, lock_key = label.split(self.WORKER_LABEL_SEPARATOR, maxsplit=1)
```
isn't really much different here